src/Block.cpp

```diff
@@ -295,6 +295,13 @@ Block::Output(std::ostream &out, FactMgr* fm, int indent) const
 	ss << "block id: " << stm_id;
 	output_comment_line(out, ss.str());
 
+	if (atomic_blk){
```

Comment:
Can you please make `rnd_flipcoin` here?

Reply:
Done. Now, instead of creating atomic block at start, they are created anywhere in a function.

Reply:
Which is much nicer!

Reply:
This one! Is pull request ready to merge?